Atas de Reuniões (#137)
UnBArqDsw2021-1/2021.1_G04_Cardeal: docs/padroesDeProjeto/atas/17-09-21.md

# Ata de Reunião 17/09/2021
## 1. Local
&emsp;&emsp; Reunião remota realizada no Microsoft Teams.

## 2. Participantes
 - Bruno Nunes
 - Douglas Castro
 - Estevão Reis
 - Giovana Vitor
 - Gustavo D.
 - Igor Sousa
 - Marcos C.
 - Pedro Haick
 - Luiz Pinheiro
 - Tomás Veloso
  
## 3. Início
 &emsp;&emsp; A reunião foi iniciada às 08:09.

## 4. Assuntos abordados
 1. Reestruturação dos documentos
 2. Aplicações dos padroes de projeto
 3. Conclusão dos dois primeiros épicos

## 5. Discussões
### 5.1. Reestruturação dos documentos

>  &emsp;&emsp; Uma discussão foi realizada para buscar reestruturar como os documentos do projeto serão exibidos. Foi decidido que haveria apenas 6 documentos principais: Documento de GRASPS, GoFs, GoFs Criacionais, GoFs Estruturais, GoFs Comportamentais e Padrões Emergentes. Cada um dos padrões serão inseridos dentro de seus respectivos documentos em vez de ser criado um documento para cada padrão especificos. Foi redefinido a responsabilidade de desenvolver cada documento, além de que foi acordado que todos os membros buscariam participar dos documentos com revisores. Esse foi o resultado da reestruturação

Reviewer: ```suggestion
 &emsp;&emsp; Uma discussão foi realizada para buscar reestruturar como os documentos do projeto serão exibidos. Foi decidido que haveria apenas 6 documentos principais: Documento de GRASPS, GoFs, GoFs Criacionais, GoFs Estruturais, GoFs Comportamentais e Padrões Emergentes. Cada um dos padrões serão inseridos dentro de seus respectivos documentos em vez de ser criado um documento para cada padrão específicos. Foi redefinido a responsabilidade de desenvolver cada documento, além de que foi acordado que todos os membros buscariam participar dos documentos com revisores. Esse foi o resultado da reestruturação
```

  * 6.Alta Coesão - Estevao
  * 7.Baixo Acoplamento - Giovana
  * 8.Indireção - Giovana
  * 9.Fabricação ou Invenção Pura - Douglas
  * 10.Variações Protegidas - Douglas
  * 11.Referências
  * 12. Versionamento 

#### 5.1.2. GoFs:

##### 5.1.2.1. Criacionais
  * 1. Introdução - Luiz
  * 2.  Factory Method - Pedro
  * 3. Abstract Factory - Igor
  * 4. Singleton - Luiz 
  * 5. Builder - Igor
  * 6. Multiton - Pedro
  * 7. Object poll - Pedro
  * 8. Prototype - Luiz
  * 9. Referências
  * 10. Versionamento 

##### 5.1.2.2. Estruturais
  * 1. Introdução - Marcos
  * 2 - Adapter - Marcos
  * 3 - Composite - Gustavo
  * 4 - Bridge - Marcos
  * 5 - Decorator - Gustavo
  * 6 - Facade - Igor
  * 7 - Flyweight - Gustavo
  * 8 - Proxy - Luiz
  * 9. Referências
  * 10. Versionamento 

##### 5.1.2.3. Comportamentais
* Introdução - Bruno
* Strategy - Bruno 
* State - Bruno
* Command - Bruno
* Iterator - Tomás
* Mediator - Tomás
* Chain of Responsability - Marcos
* Observer - Tomás
* Visitor - Marcos
* Memento - Bruno
* Template Method - Tomás
* Referências
* Versionamento 
  
#### 5.1.3. Padrões de Projeto Extras - EMERGENTES (nível de modelagem & nível de código):

>   * Introdução

Reviewer 2: ```suggestion
```

  * 1. Introdução - Estevao 
  * 2. Middlewares - Estevao
  * 3. Referências
  * 4. Versionamento 

### 5.2. Aplicações dos padroes de projeto

>   &emsp;&emsp; Buscando saber como estava a questão da aplicação do projeto, o integrante Estevão, responsável como Scrum Master no momento, perguntou a todos os integrantes se conseguiram aplicar ao menos um padrão no desenvolvimento em que cada um estava responsável. Alguns conseguiram aplicar no front end como o Singleton e o Observer. Outros disseram que estam buscando saber como aplica-los no que ja foi desenvolvido. 

Reviewer: ```suggestion
  &emsp;&emsp; Buscando saber como estava a questão da aplicação do projeto, o integrante Estevão, responsável como Scrum Master no momento, perguntou a todos os integrantes se conseguiram aplicar ao menos um padrão no desenvolvimento em que cada um estava responsável. Alguns conseguiram aplicar no front end como o Singleton e o Observer. Outros disseram que estão buscando saber como aplica-los no que já foi desenvolvido. 
```

### 5.3. Conclusão dos dois primeiros épicos

>   &emsp;&emsp; Sobre a finalização dos épicos, o pessoal responsável pelo back end relatou que quase todas as features ja estavam implementadas. No front-end a maoria das features estão na metade de sua conclusão. Todos vão bucar terminar suas tarefas até domingo dia 19/09/2021 para juntar todas as features e concluir os dois épicos

Reviewer: ```suggestion
  &emsp;&emsp; Sobre a finalização dos épicos, o pessoal responsável pelo back end relatou que quase todas as features ja estavam implementadas. No front-end a maioria das features estão na metade de sua conclusão. Todos vão buscar terminar suas tarefas até domingo dia 19/09/2021 para juntar todas as features e concluir os dois épicos
```

## 6. Conclusão

>   &emsp;&emsp; Por fim, foi acordado tudo que foi decidido na reunião, todos estão comprometidos a terminar tudo que é necessário para a proxima entrega. Foi marcando a próxima reunião para a granvação dos videos referentes à próxima entrega. A reunião foi encerrada as 9:45. 

Reviewer: ```suggestion
  &emsp;&emsp; Por fim, foi acordado tudo que foi decidido na reunião, todos estão comprometidos a terminar tudo que é necessário para a próxima entrega. Foi marcando a próxima reunião para a gravação dos videos referentes à próxima entrega. A reunião foi encerrada as 9:45. 
```

## 7. Versionamento

| Data       | Versão | Descrição         | Autores       |
| ---------- | ------ | ----------------- | ------------- |

> | 17/09/2021 | 1.0    | Adição da ata | Estevão Reis  |

Reviewer: ```suggestion
| 17/09/2021 | 1.0    | Adição da ata | Estevão Reis  |
| 17/09/2021 | 1.1    | Revisão do texto | Giovana Dionisio |
```
